src/firestarter/Environment/X86/Payload/AVX512Payload.cpp

```diff
@@ -22,13 +22,34 @@
 #include "firestarter/Environment/X86/Payload/AVX512Payload.hpp"
 #include "firestarter/Environment/X86/Payload/CompiledX86Payload.hpp"
 
+#include <asm/prctl.h> /* Definition of ARCH_* constants */
+#include <immintrin.h>
+#include <sys/syscall.h>
```

Comment:
Also include guard for this header. See the comment in the `request_permission` function